NewForce-Cohort-9/newforce-c9-tabloid-full-stack-TabloidFullStack
# Finished Deactivate logic

# Description
Added capability for an admin to be able to deactivate user profiles. 
[Trello Ticket #25](https://trello.com/c/ZQkjsBQE)

# Testing Instructions
Add and commit any unsaved changes on your branch. Run git checkout main, git fetch --all, and git checkout feat-user-profiles.
If you don't see the most recent version of my files, run git pull origin feat-user-profiles.



After these prerequisites are done, do the following:

1. In VS Community (the purple one), find your Tabloid database in the SQL Object Explorer and right click it to create a new query. 
2. Copy and paste the contents from the `03_Tabloid_Add_Col_User_Profile` SQL script included in this commit. 
3. In your query, highlight the contents and click the green button to execute the query. 
4. Run a `SELECT * FROM UserProfile` to verify that the column IsDeactivated has been added and that every row has a default value of 0 for this column (equates to the boolean false).


Afterwards, do the following: 

Inside of Visual Studio, click the execute with debugger solid green button to launch the app.
In your terminal, navigate to where your package.json file is in you client folder. Run "npm start" in your terminal.

If not logged in,  log in and use an admin account like [foo@bar.com](mailto:foo@bar.com) for the email parameter. Leave the password text box blank. Click "Login".

Navigate to User Profiles in the navbar. You should now see each user profile listed with a Deactivate button beside each account. Choose an account to deactivate and click the Deactivate button to be taken to the `/profile/deactivate/{id}` route. 

First, click the Cancel button and ensure you are redirected back to the user profile list. Afterwards, click Deactivate again. This time, when you view the confirmation, click the red Deactivate button to deactivate the account.

Your database should have updated and you should be redirected back to the user profile list. The button beside the user you deactivated should now say "Reactivate". Don't worry about this yet, as it isn't implemented.

Retrieve the email of the user that you deactivated from your database. Click Logout in the navbar, and attempt to login with the deactivate user's email address. It should fail. 


***Will update this PR with better instructions when I return 9/24 in the morning.


Checklist:
[x] My code follows the style guidelines of this project
[x] I have performed a self-review of my code
[x] I have commented my code, particularly in hard-to-understand areas
[x] My changes generate no new warnings or errors
[x] I have added test instructions that prove my fix is effective or that my feature works
